eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -178,6 +182,7 @@ data ImpTestState era = ImpTestState
   , impLastTick :: !SlotNo
   , impGlobals :: !Globals
   , impLog :: !(Doc ())
+  , impHashVerKeyVRFIdx :: !Integer
```

Comment:
Why do we need a separate index? We could just have one index for all hashes.

So, renaming `impSafeHashIdx` to `impHashIdx` and just use that.

Less state we have to keep around the better.